src/dune_rules/odoc.ml

```diff
@@ -187,9 +187,18 @@ end = struct
   let odoc_input t = t
 end
 
-let odoc sctx =
+let run_odoc sctx =
   let dir = (Super_context.context sctx).build_dir in
-  SC.resolve_program sctx ~dir "odoc" ~loc:None ~hint:"opam install odoc"
+  let open Memo.O in
+  let+ program =
+    SC.resolve_program sctx ~dir "odoc" ~loc:None ~hint:"opam install odoc"
+  in
+  let deps = Action_builder.env_var "ODOC_SYNTAX" in
+  let run ~dir args =
+    let open Action_builder.With_targets.O in
+    Action_builder.with_no_targets deps >>> Command.run ~dir program args
+  in
+  run
```

Comment:
do you mean, it should be `fun ~dir args -> ...`? works for me

Reply:
just checking - there's no particular issue with returning a function type from Memo?

Reply:
ah ok I was confused about how memo works. Sure, I'll hoist it up.

Reply:
The current code made it appear as if you're returning `run` to avoid the cost of doing resolve_program more than once. But that resolve call is memoized, so there's no obvious benefit to do so. It just seemed a little confusing to me.

Reply:
ok I get it now. How does it look now? I suppose we could factor the various `odoc_base_flags` things but I'm not sure there's a clear benefit to that.

Reply:
I wouldn't mind that change since flags sometimes change and we could easily forget to change them everywhere. It's not mandatory for this PR though.

Reply:
done. I noticed that some commands do not pass these flags (but it seems that they can't fail depending on the input); also some commands pass the input file and others the output file but only the directory is meaningful. I don't know if we need to fix this.